File: src/main/java/com/saomc/util/JString.java
package com.saomc.util;

public final class JString implements Strings {

    private final String string;

    public JString(Object object) {
        string = object instanceof String ? (String) object : String.valueOf(object);
    }

    public JString() {
        string = "";
    }

    @Override
    public final String toString() {
        return string;
    }

}

File: src/main/java/com/saomc/saoui/util/JString.kt
package com.saomc.saoui.util

class JString : Strings {

    private val string: String

    constructor(`object`: Any) {
        string = `object` as? String ?: `object`.toString()
    }

    constructor() {
        string = ""
    }

    override fun toString(): String {
        return string
    }

}
